### PR TITLE
add theme and language selections in user menu

### PR DESCRIPTION
**Pull request by abhinavrastogi-harness**
Theme selector:
Screenshot 2024-12-13 at 3 58 38 PM

Language selector:
Screenshot 2024-12-13 at 4 11 29 PM

### CODE REVIEW

File: packages/ui/src/views/layouts/SandboxRoot.tsx [hunk 1]
```diff
@@ -22,13 +21,14 @@ export interface SandboxRootProps {
   currentUser: TypesUser | undefined
   logout?: () => void
   useNav: () => NavState
-  t: TFunction
+  useThemeStore: () => IThemeStore
+  useTranslationStore: () => TranslationStore
 }
 
-export const SandboxRoot = ({ currentUser, useNav, t, logout }: SandboxRootProps) => {
+export const SandboxRoot = ({ currentUser, useNav, logout, useThemeStore, useTranslationStore }: SandboxRootProps) => {
```
Comment (knagurski) [inline packages/ui/src/views/layouts/SandboxRoot.tsx:28]: Looks like this causes some type issues in the project

File: packages/ui/src/components/navbar/navbar.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { useMemo } from 'react'
 import { Link, useLocation, useNavigate } from 'react-router-dom'
 
-import { Button, Icon, NavbarProjectChooser, ScrollArea, Spacer } from '@/components'
+import { Button, Icon, IThemeStore, NavbarProjectChooser, ScrollArea, Spacer } from '@/components'
 import { TypesUser } from '@/types'
-import { TFunction } from 'i18next'
+import { TranslationStore } from '@views/index'
```
Comment (knagurski) [inline packages/ui/src/components/navbar/navbar.tsx:6]: I had a PR open to allow import from `@components` and `@views`, but it got out of date. Will open a new one, but in the meantime, `@/views` is probably a bit easier. Not a blocker in any way

File: packages/ui/src/components/navbar/navbar.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import { useMemo } from 'react'
 import { Link, useLocation, useNavigate } from 'react-router-dom'
 
-import { Button, Icon, NavbarProjectChooser, ScrollArea, Spacer } from '@/components'
+import { Button, Icon, IThemeStore, NavbarProjectChooser, ScrollArea, Spacer } from '@/components'
```
Comment (knagurski) [inline packages/ui/src/components/navbar/navbar.tsx:4]: That bit of Hungarian notation really stands out as being different. Will open a ticket to start standardising on naming.

**Review by srdjan-harness — APPROVED** [review]

File: packages/ui/src/views/layouts/SandboxRoot.tsx (hunk 1 above)
Comment (abhinavrastogi-harness) [reply packages/ui/src/views/layouts/SandboxRoot.tsx:28]: yup, checking this

**Pull request merged**